src/rust/engine/src/context.rs

```diff
@@ -66,11 +66,22 @@ impl Core {
         .unwrap_or_else(|e| {
           panic!("Could not initialize VFS: {:?}", e);
         }),
-      pool: RwLock::new(Some(Core::create_pool())),
+      pool: RwLock::new(None),
     }
   }
 
   pub fn pool(&self) -> RwLockReadGuard<Option<CpuPool>> {
+    {
+      let pool = self.pool.read().unwrap();
+      if pool.is_some() {
+        return pool;
+      }
+    }
+    // If we get to here, the pool is None and needs re-initializing.
```

Comment:
This could race with some other initializing thread, resulting in multiple pools getting created and thrown away.

Need to double check inside the write lock that it hasn't already been intialized.